manta-pay/src/test/signer.rs

```diff
@@ -281,3 +281,63 @@ async fn find_the_bug() {
         wallet.signer_mut().prune();
     }
 }
+
+/// Tests [`consolidate`](manta_accounting::wallet::Wallet::consolidate) works as expected.
+#[ignore] // We don't run this test on the CI because it takes a long time to run.
+#[tokio::test]
+async fn consolidation_test() {
```

Comment:
I didn't add them because this is super-safe already. If you somehow managed to create an invalid consolidation request the resulting transfer would be invalid, e.g., you try to spend the same asset twice (duplicates in the request), you try to spend assets with different IDs (transfer protocol always rejects that), you try to spend assets you don't own (the `check` function returns `false`).
The only ones that would pass are those which involve zero assets (which besides being stupid and costing gas won't do any harm, and why would you select zero assets in the first place?), including the consolidation request with a single element.